xapian-letor/configure.ac

```diff
@@ -34,10 +34,30 @@ dnl 0:0:0 1.3.1 First release with xapian-letor split out from xapian-core
 LIBRARY_VERSION_INFO=0:0:0
 AC_SUBST(LIBRARY_VERSION_INFO)
 
+LIBRARY_VERSION_SUFFIX=
+dnl Where xapian.h, etc go.
+case $PACKAGE_VERSION in
+  [*.*[13579].*])
+    dnl Development release series:
+    [dev_series=`echo "$PACKAGE_VERSION"|sed 's/\.[^.]*$//'`]
+
+    dnl Append version installed programs by default.  To specify no suffix,
+    dnl configure with: --program-suffix=
+    test x"$program_suffix" != xNONE || program_suffix=-$dev_series
+
+    LIBRARY_VERSION_SUFFIX=-$dev_series
+    ;;
+esac
+
+AC_SUBST([LIBRARY_VERSION_SUFFIX])
+
 dnl Where xapian-letor.h, etc go.  In development release append "/xapian-1.3".
 incdir=$includedir
 AC_SUBST([incdir])
```

Comment:
This comment doesn't seem to match the actually behaviour.  Comparing the code with `xapian-core/configure.ac` the code to add there's no equivalent of `incdir=$incdir/xapian-$dev_series` here, which is what implements that behaviour in development releases.

The reason we qualify the include directory paths like this is to ease parallel installation of stable and development versions.  Currently that's not a big concern for letor as there is no stable version.  So either we can make this match core and qualify the path (for future-proofing) or we make the comments match what actually happens here.

Reply:
Here my assumption was that anything which uses letor would eventually also need to use core and hence I am using `LIBRARY_VERSION_SUFFIX` which does not indicate letor version but core version which in turn is substituted in xapianletor-config script as I am adding libxapian-1.5.so and libxapianletor.so to the --ltlibs option in xapianletor-config.in script. Here probably we can change the name to something like `XAPIAN_LIBRARY_VERSION_SUFFIX` as the current name is misleading. Please correct me if I am wrong.

Reply:
When using the `--ltlibs` option libtool will take care of ensuring the appropriate libxapian gets linked in to any executable linked against xapian-letor's library - either the platform handles this automatically (as on Linux for example), or libtool will add libraries listed in the `dependency_libs` entry in `libxapianletor.la` (which is just a text file if you want to take a look at it).

It also seems wrong for xapian-letor to assume what the suffix on libxapian is.  There's no inherent reason why you need to use matching versions of xapian-core and xapian-letor.  I think xapian-letor should really behave just like anything else using the xapian-core API and use `XO_LIB_XAPIAN`/`xapian-config` (or similar) to find out the compiler options to use for linking against xapian-core - these will include any suffix needed.

Reply:
understood fixed in a fixup commit.